### ISSUE TITLE
multiple card support

### ISSUE THREAD

**Issue by rakslice** [rakslice/leaix#3]
This would need work to properly support multiple cards. It is totally untested. There is no disambiguation for interrupts, so an interrupt on any card is treated as an interrupt on each of the cards -- I never figured out if there was some state already accessible to the interrupt handler that would allow for this or if I needed to implement some cheese to provide callback functions at different addresses for each card.

**Comment by rakslice** [comment]
In AIX PS/2, the headers don't use the correct types for any function pointer parameters; you have to consult the Technical Reference to find the correct signatures for any individual callback functions.  In this case  `ddintr()` (the interrupt handler you register with `intrattach()`) is called with a value `int vec_num` which is the interrupt number it is being called for, which if you've registered the same handler for multiple interrupts disambiguates which registration the call is about.

tl;dr: The idea is that you have the interrupt values somewhere global (in our case already in `ISA_ATTACH_ARGS(unit)->ia_irq`) and then in your interrupt handler you use this info to work which device you were called about.

So bumping up `NUM_INTERFACES` and then drilling a couple of holes to pass the `vec_num` to `leintr()` and then modifying the logic in it to check if the unit's `ia_irq` matches that value and only call `leintr_unit` if so, should get us started.

I don't know what other gotchas with multiple cards might have crept into the codebase; it doesn't seem the actual value of `NISA` matters as long as it is nonzero.

**Comment by rakslice** [comment]
Ah, the initial pci card enumeration logic needs to be updated to keep looking for additional cards up to the configured maximum I guess.
